### ISSUE TITLE
When you call `python train.py --deploy --eval <-other-commands->` the network is pruned at runtime (by the `if` statement on line 73) and then used.

### ISSUE THREAD

**Issue by cvJie** [BayesWatch/pytorch-prunes#2]
When you call `python train.py --deploy --eval <-other-commands->` the network is pruned at runtime (by the `if` statement on line 73) and then used.  

If you would like to save the pruned model, you could put a:
```
save_checkpoint({
                'epoch': SD['epoch'],
                'state_dict': model.state_dict(),
                'error_history': SD['error_history'],
            }, filename=filename)
```

after the call to `pruner.compress(model)` on line 89. 

However, it is better to leave the saved checkpoint in its original state, with the masks attached, to make it easier to recover the pruning trajectories. You can check the number of ops/params with `get_inf_params(model)` if you want an indication of compression rate. Whenever you want to use a pruned model for something, you then just need to run the code from lines 71 to 89 in `train.py`, followed by the task you want the pruned model to complete. 

Hope this helps.

_Originally posted by @jack-willturner in https://github.com/BayesWatch/pytorch-prunes/issues/1#issuecomment-541559480_
